### PR TITLE
OCPBUGS-28650: fix generation of telemeter token hash

### PR DESCRIPTION
**Pull request by simonpasquier**
* [ ] I added CHANGELOG entry for this change.
* [X] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: pkg/manifests/manifests.go [hunk 1]
```diff
@@ -3076,8 +3076,9 @@ func (f *Factory) TelemeterClientDeployment(proxyCABundleCM *v1.ConfigMap, s *v1
 
 	// Set annotation on deployment to trigger redeployments
 	if s != nil {
-		hash := sha256.New()
-		d.Spec.Template.Annotations["telemeter-token-hash"] = string(hash.Sum(s.Data["token"]))
+		h := fnv.New64()
```
Comment (danielmellado) [inline pkg/manifests/manifests.go:3079]: Can we have some quick note about the motivation for the change betweeh sha256 and the fnv hash? Thanks!

Comment (simonpasquier) [reply pkg/manifests/manifests.go:3079]: added a description in the commit message.

**Review by rexagod — APPROVED** [review]
IIUC even though we could `strconv.ParseUint(<telemeter-token-hash>, 32, 64)` to get back the generated hash (`h.Sum64()`), that shouldn't matter as it'd be different from the actual token hash in `pull-secret` since the hashing algorithm being used here is different.

**Pull request merged**